src/common/feedGenerator.ts

```diff
@@ -536,9 +536,15 @@ export const sourceFeedBuilder = (
 ): SelectQueryBuilder<Post> => {
   builder.andWhere(`${alias}.sourceId = :sourceId`, { sourceId });
 
-  if (sourceId === 'community') {
-    builder.andWhere(`${alias}.banned = false`);
-  }
+  builder.andWhere(
+    new Brackets((qb) => {
+      return qb
+        .where(`${alias}.authorId = :userId OR ${alias}.scoutId = :userId`, {
+          userId: ctx.userId,
+        })
+        .orWhere(`${alias}.banned = false AND ${alias}.showOnFeed = true`);
+    }),
+  );
```

Comment:
@idoshamun just need a sanity check. Would this be enough? It should filter out banned posts unless you're author or scout.

Reply:
Same from @capJavert if you could do a sanity check 🙈

Reply:
Could maybe add this to `Post` on graphorm too.

Reply:
```ts
qb
  .andWhere(`"${alias}"."deleted" = false`)
  .andWhere(`"${alias}"."visible" = true`)
  .andWhere(
    new Brackets((bqb) => {
      return bqb
        .where(
          `${alias}.authorId = :userId OR ${alias}.scoutId = :userId`,
          {
            userId: ctx.userId,
          },
        )
        .orWhere(
          `${alias}.banned = false AND ${alias}.showOnFeed = true`,
        );
    }),
  ),
```

Though not 100% if it's side-effect free

Reply:
I think it should work, but it does remove posts on source feed (i.e. `bit`) when the post has `banned` set.

Reply:
Query looks good, If we want to apply it globally adding it in graphorm makes sense to me. As to `banned` thing, let's see what @idoshamun says since I am not sure if it was due to any decision.

Reply:
Let's not add it to the post because then the post page will not be available and this is not what we want

Reply:
Ah yeah, so I'll adding it to `sourceFeedBuilder` is probably best location.

Reply:
I went with adding `vordr` flag, as it would block Welcome posts, and I didn't really feel like adding a complex query just to catch edge cases.
It still adds the other flags as before, just for filtering purposes, I added the vordr flag.

Reply:
That way it shouldn't change anything how it works today